proposals/new-project-addition-process.md

```diff
@@ -0,0 +1,36 @@
+## Motivation
+As Kubeflow has become a well established ecosystem and community, several
+projects may want to be incorporated into Kubeflow in different capacities.
+
+That may include:
+1. Project Donation
+2. Inclusion in the Manifest
+3. Inclusion in documentation as an add on
```

Comment:
I think the process should probably be the same but the requested outcome should be different. 

I think really we're just formalizing what standards we have for Feast and KServe today and I'd like to be able to include Feast into the manifest with this process. That is my personal goal.

Reply:
From my point of view, the proposal should be dedicated to all future projects that want to join Kubeflow ecosystem. As we discussed before, we might want to re-consider adding external add-ons to the manifests repo, since the manifests repo should be an aggregation of Kubeflow components.

Reply:
@andreyvelich @franciscojavierarceo I want get some clarification, when you mention we need to include/or not  manifests, do you guys mean as single package manifests that include core + add-ons or you are thinking both should be different repos and add-ons not even present in separate package in manifests repo?

Reply:
>I view Kubeflow as the best open-source MLOps platform that not only offers a community sharing best practices but also provides all the core components needed to build a successful ML platform.

Given the size of AI and cloud native landscapes and number of different ML use-cases, how can we agree on what should be included in Kubeflow platform ?

>And I think that's ultimately why some add-ons should be included in the manifest.

Can you explain the motivation for the Kubeflow platform to include those add-ons ?

If some Kubeflow components have integrations with other projects, they can always add user instructions on how to use it. For example, Katib Trials can orchestrate Tekton Pipelines (that we don't install in Kubeflow manifests) and we have instruction for it: https://github.com/kubeflow/katib/tree/master/examples/v1beta1/tekton.

Reply:
> do you guys mean as single package manifests that include core + add-ons or you are thinking both should be different repos and add-ons not even present in separate package in manifests repo?

@rareddy From my personal perspective, Kubeflow manifests should not have add-ons folder since it is out-of-scope for Manifests WG: https://github.com/kubeflow/community/blob/master/wg-manifests/charter.md#code-binaries-and-services

Reply:
>If we do not include open source tools and frameworks in the ecosystem (i.e., the deployment) then I'm not sure we being sincere in that statement.

I think, this statement is explaining that we have deep integration with open-source AI/ML projects like PyTorch, Jax, TensorFlow, Triton, etc. and provide a way to use them on Kubernetes. Basically, it is top part of this diagram: https://www.kubeflow.org/docs/started/architecture/#kubeflow-ecosystem.


Additionally, how can we deal with similar projects ? For example, we have Argo, Tekton, Airflow for workflow engine, and what should we add to Kubeflow ecosystem ?

Reply:
>I think, this statement is explaining that we have deep integration with open-source AI/ML projects like PyTorch, Jax, TensorFlow, Triton, etc. and provide a way to use them on Kubernetes. Basically, it is top part of this diagram: https://www.kubeflow.org/docs/started/architecture/#kubeflow-ecosystem.

That's fair point. 

>Additionally, how can we deal with similar projects ? For example, we have Argo, Tekton, Airflow for workflow engine, and what should we add to Kubeflow ecosystem ?

I think this is a good area for discussion. Here's my opinion: 

I think we should have a set of recommended Kubeflow add-ons (i.e., components) that complete the "MLOps platform" and ideally users contribute to and use them. 

1. If a component doesn't exist, we should consider adding it as an add-on. If there's a gap in an existing component, maintainers/contributors would address it. 
2. If the gap is not addressed and a better solution emerges that makes the existing component inferior, we should consider favoring a newly proposed component over an existing one (as this would likely be a symptom of an unresponsive or dormant community). 

I think (1) is what helps create a full platform and (2) is what helps ensure that we do what's best for the community.

Argo, Tekton, and Airflow are an interesting case. I think having the component/project make the decision is probably what's best.

Reply:
>Argo, Tekton, and Airflow are an interesting case. I think having the component/project make the decision is probably what's best.

Who is going to make a decision on what components we should promote/recommend as add-ons ? And if there is a gap in existing Kubeflow component, why just not improve it to address this gap ?

Argo, Tekton, Airflow is not a single example. There are many other projects that have overlaps: Ray and Training Operator/KServe/Katib, Volcano vs Training Operator, Kaito vs KServe, MLFlow vs Model Registry, etc.
Eventually, we have users who just use native Kubernetes workloads for MLOps/LLMOps.

Reply:
>Who is going to make a decision on what components we should promote/recommend as add-ons ? And if there is a gap in existing Kubeflow component, why just not improve it to address this gap ?

I think we should take input from the community! And part of that means addressing the gap! At the moment, Kubeflow doesn't cover everything, so I think there are two conversations here and it's worth disambiguating between them in my opinion.

For existing components, we should enhance but for missing components we should fill the gap with either an Add-on or a new project. Feast is a good example here. Would you want to make a Kubeflow Feature Store or use Feast?

Reply:
>Argo, Tekton, Airflow is not a single example. There are many other projects that have overlaps: Ray and Training Operator/KServe/Katib, Volcano vs Training Operator, Kaito vs KServe, MLFlow vs Model Registry, etc.
Eventually, we have users who just use native Kubernetes workloads for MLOps/LLMOps.

Yeah, I agree that there are many examples for sure. I also think that we should aim to solve for like 80% of use-cases and not 20% who may always inevitably customize. In my opinion, having 80% of use cases solved by users is an impactful thing. Many organizations will benefit from this and that will help foster a healthy community.

Reply:
>Would you want to make a Kubeflow Feature Store or use Feast?

I agree that it is always better to consolidate effort to solve the specific problem instead of building something new.
Unfortunately, it is hard to consolidate effort between cloud native, AI/ML, and Kubeflow communities.

@franciscojavierarceo What are your thoughts on the conformance program for Kubeflow distribution in that case?
When distribution can call themself **Kubeflow Certified** ?

Reply:
I think that would make a lot of sense. I think in order to be included into the manifest the Add-on/component/product should obviously work easily and I think it'd be the obligated of the add-on project to keep the project moving forward and working with Kubeflow or face removal.

Reply:
I assume you're suggesting a conformance program similar to what's available for Kubernetes (https://github.com/cncf/k8s-conformance/tree/master)

Reply:
Yeah, my question is how do you think we should design this program for distributions if we have add-ons in Kubeflow ecosystem ?
Should every Kubeflow distribution support all add-ons and who will create the test suites for those add-ons ?